controller/src/test/java/io/pravega/controller/server/eventProcessor/ControllerEventProcessorsTest.java

```diff
@@ -90,13 +91,57 @@ public void testEventKey() {
         assertEquals(commitEvent.getKey(), "test/test");
     }
 
+    @Test(timeout = 10000)
+    public void testIsReady() {
+        Controller localController = mock(Controller.class);
+        CheckpointStore checkpointStore = mock(CheckpointStore.class);
+        StreamMetadataStore streamStore = mock(StreamMetadataStore.class);
+        BucketStore bucketStore = mock(BucketStore.class);
+        ConnectionPool connectionPool = mock(ConnectionPool.class);
+        StreamMetadataTasks streamMetadataTasks = mock(StreamMetadataTasks.class);
+        StreamTransactionMetadataTasks streamTransactionMetadataTasks = mock(StreamTransactionMetadataTasks.class);
+        KVTableMetadataStore kvtStore = mock(KVTableMetadataStore.class);
+        TableMetadataTasks kvtTasks = mock(TableMetadataTasks.class);
+        ControllerEventProcessorConfig config = ControllerEventProcessorConfigImpl.withDefault();
+        EventProcessorSystem system = mock(EventProcessorSystem.class);
+        CuratorZookeeperClient curatorZKClientMock = mock(CuratorZookeeperClient.class);
+        CuratorFramework client  = mock(CuratorFramework.class);
+        when(checkpointStore.isHealthy()).thenReturn(true);
+        @Cleanup
+        ControllerEventProcessors processors = spy(new ControllerEventProcessors("host1",
```

Comment:
Why are we spying this object? We should not spy the object we intend to test.
Please look at code in ControllerServiceStarter.startup() method, that invokes the bootstrap() method on CEP. Only after this method has completed successfully isBootStrapCompleted() will be true....

Reply:
Since our aim to test isReady() method only, so I'm using spy. This way it's easier to test the functionality.

Reply:
It may be better to use the actual object instead of a spy here, as the test is for CEP.

Reply:
changes made as per discussion